lib/security/aes.js

```diff
@@ -0,0 +1,42 @@
+var CryptoJS = require("crypto-js");
+var q = require('q');
+
+/**
+* Encrypt an Object using AES with a specified secret key.
+* @param data - the data to encrypt.
+* @param key - the secret key to encrypt with.
+*/
+function encrypt(data, key) {
+  var deferred = q.defer();
```

Comment:
The Q module follows the Promise spec, so while it's as feature-full as Bluebird, it's not bad to use if you don't find yourself needing more complex stuff like Promise.map() or Promise.fromCallback(). (I'd still recommend Bluebird if you can add it.)

Q still has a way around using deferreds.
```js
return Q.Promise((resolve, reject) => ...);
```
The reason to avoid deferreds is because if you get an unexpected error in the code (such as if JSON.parse fails down on line 33), it will throw an exception, but that exception will not be caught by anything. Q.Promise (as well as Promise constructor for other libraries) automatically wraps the resolver function argument in a try/catch and will propagate all errors.